### PR TITLE
store emitter keep source vec values intact

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
 - *store emitter keep source vec values intact*
 - onednn PR: https://github.com/openvinotoolkit/oneDNN/pull/180

### Tickets:
 - *105903*

### CODE REVIEW

File: src/plugins/intel_cpu/src/emitters/jit_load_store_emitters.cpp [hunk 1]
```diff
@@ -641,23 +640,25 @@ void jit_store_emitter::emit_isa(const int in_vec_idx, const Xbyak::Reg64 &reg_d
     }
     using Vmm = typename conditional3<isa == cpu::x64::sse41, Xmm, isa == cpu::x64::avx2, Ymm, Zmm>::type;
 
-    int data_idx = in_vec_idx;
+    original_data_idx = in_vec_idx;
```
Comment (IvanNovoselov) [inline src/plugins/intel_cpu/src/emitters/jit_load_store_emitters.cpp:643]: As far as I understood, the primary purpose of `original_data_idx` is to indicate that `data_idx` will be corrupted and it's update is required ( numerous `if (data_idx == original_data_idx)` statements). So why don't we store a boolean variable with a descriptive name instead? like `data_reg_update_required`? Then all the `if` statements will look like `if (data_reg_update_required)` which is more readable. What do you think?

Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/emitters/jit_load_store_emitters.cpp]: I agree, have updated, thank you!

File: src/plugins/intel_cpu/src/emitters/jit_load_store_emitters.cpp [hunk 2]
```diff
@@ -742,14 +752,26 @@ void jit_store_emitter::store_bytes(const Vmm &vmm, const Xbyak::Reg64 &reg, int
...
             start_bytes += 32;
             bytes_to_store -= 32;
-            h->vextractf64x4(ymm, zmm, 1); // load upper bits from zmm into ymm
+            // load upper bits from zmm into ymm
+            if (data_idx == original_data_idx) {
+                h->vextractf64x4(Ymm(aux_src_idx), zmm, 1);
+                update_data_idx();
+            } else {
+                h->vextractf64x4(ymm, zmm, 1);
+            }
```
[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/emitters/jit_load_store_emitters.cpp]: This is a good idea, have updated, thank Ivan!

**Comment by dmitry-gorokhov** [comment]
@a-sidorova please take a look as well

**Review by IvanNovoselov — APPROVED** [review]
Looks good, no further suggestions.

**Review by a-sidorova — APPROVED** [review]
LGTM

But need to update PR to fork (to resolve merge conflicts)

**Comment by chenhu-wang** [comment]
> LGTM
> 
> But need to update PR to fork (to resolve merge conflicts)

done, thanks!

File: src/plugins/intel_cpu/src/emitters/jit_load_store_emitters.cpp [hunk 3]
```diff
@@ -15,6 +15,19 @@ using namespace dnnl::impl::cpu::x64;
 using namespace Xbyak;
 using namespace Xbyak::util;
 
+#define STORE_KEEP_SOURCE(is_updated, instruction, data_reg, data_reg_new, ...) \
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/emitters/jit_load_store_emitters.cpp:18]: I would say this marco makes the code mich less readable. Can we at least add some comments that descibe its behavior. Also I wouls suggest to get rid of `is_updated` argument given it is always equal to `data_reg_updated`

**Comment by dmitry-gorokhov** [comment]
@chenhu-wang Please rebase on latest master as well

File: src/plugins/intel_cpu/src/emitters/jit_load_store_emitters.cpp (hunk 3 above)
Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/emitters/jit_load_store_emitters.cpp]: What about `is_updated` argument?

Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/emitters/jit_load_store_emitters.cpp]: I replaced macro parameter name from `is_updated `to `data_reg_updated ` along previous commit. I think I got your point now that remove this parameter at all from its signature. Done in [dd027ac](https://github.com/openvinotoolkit/openvino/pull/16313/commits/dd027ac791fdb5f975f798ce6f14e29d1cf3762b), take a look again please, thank you!

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
@chenhu-wang I see that PR inside the oneDNN fork contains 3 commits. Please squash it into one to keep the history clean

**Comment by chenhu-wang** [comment]
> @chenhu-wang I see that PR inside the oneDNN fork contains 3 commits. Please squash it into one to keep the history clean

@dmitry-gorokhov , done, thank you!

**Pull request merged**